src/ucs/sys/event_set.c

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (C) Hiroyuki Sato. 2019. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "event_set.h"
+
+#include <ucs/debug/memtrack.h>
+#include <ucs/debug/log.h>
+#include <ucs/debug/assert.h>
+
+#include <sys/epoll.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#define UCS_EVENT_EPOLL_MAX_EVENTS      16
+#define UCS_EVENT_SET_EPOLL_SIZE 2
+
+struct ucs_event_set_type_table {
+    int event_type;
+    ucs_event_set_type_t os_event;
+};
+
+/* int ucs_event_set_events_map(int events) */
+static int ucs_event_set_events_map(int events)
+{
+    int ret = 0;
+    int i;
+    int matched = 0;
+    struct ucs_event_set_type_table event_table[UCS_EVENT_SET_EPOLL_SIZE] = {
+      { UCS_EVENT_SET_EVREAD,  EPOLLIN  },
+      { UCS_EVENT_SET_EVWRITE, EPOLLOUT },
+    };
+
+    for (i = 0; i < UCS_EVENT_SET_EPOLL_SIZE; i++) {
```

Comment:
since there are only 2 events really, this looks like overkill, also takes more processing - just check for EPOLLIN and EPOLLOUT with 2 if statements

Reply:
I worried about additional event conversion. But, now there are two events only.  So, I used multiple if statement.